src/OpenLoco/src/Map/MapGenerator/OriginalTerrainGenerator.cpp

```diff
@@ -86,16 +86,16 @@ namespace OpenLoco::World::MapGenerator
         int32_t x = randX;
         for (auto j = 0; j < featureWidth; ++j)
         {
-            int32_t y = (featureHeight + randY - 1) & 0x1FF;
+            int32_t y = (featureHeight + randY - 1) % (heightMap.height - 1);
```

Comment:
there seems to be lots of places we are using heightMap.height - 1. Can you think of a better design so we don't have to use it as often. Not really for this PR though.